#19

Redsart opened the pull request.

Redsart committed:
Resolve some of the analyzer's errors.

Redsart committed:
Resolve CA1303: Do not pass literals as localized parameters and CA1305: Specify IFormatProvider analyzer warning's.

kpau commented on TodoApp.ConsoleApp/Services/TodoService.cs:
```diff
@@ -30,21 +30,30 @@ public Task GetByID(Guid id)
 
         public bool Update(Task task)
```
Any argument validation should be put at the top of the method, and the error handling should be done there. This way the method depth is less (one less `if` in this case).
`ArgumentNullException` (and every other Argument exception) require argument name as first parameter, not the error message. Use the [nameof operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/nameof).

Not sure why you added the explicit argument when calling `Delete`. You can skip it.

For Example:
```csharp
public bool Update(Task task)
{
    if (task == null)
    {
        throw new ArgumentNullException(nameof(task));
    }

    bool isDeleted = Delete(task.ID);
            
    if (!isDeleted)
    {
        return false;
    }

    var newTask = Create(task);
    return newTask != null;
}
```

kpau commented on TodoApp.ConsoleApp/UI/TaskOperations.cs:
```diff
@@ -90,15 +90,17 @@ static void TaskMaker()
             string title = UserInput.ReadText("Enter a title: ", true);
             string message = UserInput.ReadText("Enter a description: ", true);
-            int deadLine = int.Parse(UserInput.ReadText("How many days you need to finish the task: ",true));
+#pragma warning disable CA1305 // Specify IFormatProvider
```
Don't suppress the warning. Add `CultureInfo.CurrentCulture` as a second argument for `int.Parse()`.

kpau commented on TodoApp.ConsoleApp/UI/UserComments.cs:
```diff
@@ -0,0 +1,47 @@
+﻿namespace TodoApp.ConsoleApp.UI
+{
+    static class UserComments
```
Rename the class. `UseComments` sounds like we are commenting the user or he is giving some comments.
Examples: `Messages`, `Output`, `Text`, etc.

kpau commented on TodoApp.ConsoleApp/UI/UserInput.cs:
```diff
@@ -1,11 +1,13 @@
     {
         const string Yes = "yes";
         const string No = "no";
+        static readonly IFormatProvider provider = CultureInfo.CurrentCulture;
```
Set these 3 fields explicitly `private`.

kpau commented on TodoApp.ConsoleApp/UI/UserInput.cs:
```diff
@@ -35,6 +37,10 @@ public static string ReadText(string question, bool required = false, string def
         public static int ReadOption(string question, string[] availableOptions, bool required = false, int defaultValue = 0)
         {
             Console.WriteLine(question);
+            if (availableOptions == null)
```
Move this validation at the top of the method.

`ArgumentNullException` requires the argument name as first parameter. use `nameof` operator to provide it.

Redsart committed:
Resolve the warning's comming from the analyzer.

Redsart committed:
Resolve the coments from the revu.

kpau commented on TodoApp.ConsoleApp/UI/UserComments.cs:
```diff
@@ -0,0 +1,47 @@
+﻿namespace TodoApp.ConsoleApp.UI
+{
+    static class Messages
```
Rename the file too.

kpau commented on CustomDictionary.xml:
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
This file is duplicated. Remove the one that is not needed.

kpau commented on TodoApp.ConsoleApp.csproj:
```diff
@@ -0,0 +1,106 @@
+  <ItemGroup>
+    <None Include="App.config" />
+    <None Include="packages.config" />
+    <None Include="TodoApp.ConsoleApp.pfx" />
+    <None Include="TodoApp.Library.pfx" />
```
These two file are not pushed to the git repository.
`TodoApp.Library.pfx` should be in the Library project.

Please commit and push them.

Redsart committed:
Change the name of the Messages class to Messages, modify the .gitignore file to allow pushing pfx of  files.

Redsart committed:
Delete .pfx files from the project and add .pfx files to .gitignore

kpau committed:
#19: Code analyzer config improvements

- Run code analysis on build
- Exclude CA2210 rule (assembly strong name)
- CustomDictionary.xml Build Action: Code Analysis Dictionary
